modules/stochastic_tools/src/reporters/SingularTripletReporter.C

```diff
@@ -0,0 +1,106 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SingularTripletReporter.h"
+
+registerMooseObject("StochasticToolsApp", SingularTripletReporter);
+
+InputParameters
+SingularTripletReporter::validParams()
+{
+  InputParameters params = GeneralReporter::validParams();
+  params += MappingInterface::validParams();
+
+  params.addClassDescription("Tool for accessing and outputting the singular triplets of a "
+                             "singular value decomposition in PODMapping.");
+
+  params.addRequiredParam<UserObjectName>(
+      "pod_mapping", "The PODMapping object whose singular triplets should be printed.");
+
+  params.addRequiredParam<std::vector<VariableName>>(
+      "variables", "The names of the variables whose SVD should be printed.");
+
+  return params;
+}
+
+SingularTripletReporter::SingularTripletReporter(const InputParameters & parameters)
+  : GeneralReporter(parameters),
+    MappingInterface(this),
+    _variable_names(getParam<std::vector<VariableName>>("variables")),
+    _left_singular_vectors(
+        declareValueByName<std::map<VariableName, std::vector<DenseVector<Real>>>>(
+            "left_singular_vectors", REPORTER_MODE_ROOT)),
+    _right_singular_vectors(
+        declareValueByName<std::map<VariableName, std::vector<DenseVector<Real>>>>(
+            "right_singular_vectors", REPORTER_MODE_ROOT)),
+    _singular_values(declareValueByName<std::map<VariableName, std::vector<Real>>>(
+        "singular_values", REPORTER_MODE_ROOT))
```

Comment:
I don't think that will work. The `to_json` functions don't like pointers.  Maybe I am missing something but I think this will be printed only a few times (mostly once) in a run. I can add a warning that this is an expensive operation. I am also more or less convinced that copying this data will take less time than printing it into the json file.